hs/Database/Tables.hs

```diff
@@ -183,29 +162,19 @@ data Course =
 instance ToJSON Course
 instance ToJSON Session
 
-instance ToJSON Lecture where
-  toJSON (Lecture lectureExtra lectureSection lectureCap lectureTimeStr lectureTime lectureInstructor lectureEnrol lectureWait)
-          = object ["extra" .= lectureExtra,
-                    "section" .= lectureSection,
-                    "cap" .= lectureCap,
-                    "time_str" .= lectureTimeStr,
-                    "time" .= map convertTimeToString lectureTime,
-                    "instructor" .= lectureInstructor,
-                    "enrol" .= lectureEnrol,
-                    "wait" .= lectureWait
-                   ]
-
-instance ToJSON Tutorial where
-  toJSON (Tutorial Nothing tutorialTimes tutorialTimeStr) =
-      Array $ V.fromList [toJSON (map convertTimeToString tutorialTimes), toJSON tutorialTimeStr]
-  toJSON (Tutorial (Just value) tutorialTimes tutorialTimeStr) =
-      Array $ V.fromList [toJSON value, toJSON (map convertTimeToString tutorialTimes), toJSON tutorialTimeStr]
+instance ToJSON Time where
+    toJSON (time) =
+        toJSON $ convertTimeToString time
+
+-- instance FromJSON required so that tables can be parsed into JSON,
+-- not necessary otherwise.
+instance FromJSON Time
```

Comment:
Hmm. Does it make much sense to 1) use a Double as the representation of a time, and 2) have this discrepancy between the way that we store data and the way that we serve that data - perhaps it would be better to make the conversion at the parsing level. What are your thoughts?